http_backend_test.go

```diff
@@ -0,0 +1,94 @@
+package colly
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net/http"
+	"net/http/cookiejar"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"sync"
+	"testing"
+	"time"
+)
+
+func TestHTTPBackendDoCancelation(t *testing.T) {
+	rand.Seed(time.Now().Unix())
```

Comment:
It's better to not have any indeterminate behaviour in tests. Either just make make `p`, `n`, and `c` some constants, or, if you're unsure, use `math/rand.New()` with some constant seed and run your test multiple times using the same generator.

Reply:
Made a commit with a hardcoded seed. :)

Reply:
I'd still advise you to use `rand.New()`. Less global state - the better.